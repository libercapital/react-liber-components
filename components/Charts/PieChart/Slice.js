"use strict";

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es6.object.assign");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.array.index-of");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.object.keys");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _d3Shape = require("d3-shape");

var _styles = require("./styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Slice = function Slice(_ref) {
  var innerRadius = _ref.innerRadius,
      radius = _ref.radius,
      events = _ref.events,
      abs = _ref.abs,
      props = _objectWithoutProperties(_ref, ["innerRadius", "radius", "events", "abs"]);

  var pathFunctionBg = (0, _d3Shape.arc)().outerRadius(2 * abs + radius).innerRadius(innerRadius);
  var pathFunctionFg = (0, _d3Shape.arc)().outerRadius(abs / 2 + radius).innerRadius(innerRadius - abs / 2);
  var defaultPathFunction = (0, _d3Shape.arc)().outerRadius(radius).innerRadius(innerRadius);
  return _react["default"].createElement(_react.Fragment, null, _react["default"].createElement("g", events, _react["default"].createElement(_styles.Slice, _extends({}, props, {
    style: {
      fill: 'transparent'
    },
    pathFunction: pathFunctionBg
  })), _react["default"].createElement(_styles.Slice, _extends({}, props, {
    pathFunction: abs > 0 ? pathFunctionFg : defaultPathFunction
  }))));
};

Slice.propTypes = {
  slice: _propTypes["default"].objectOf(_propTypes["default"].any),
  // eslint-disable-line react/require-default-props
  style: _propTypes["default"].objectOf(_propTypes["default"].any),
  // eslint-disable-line react/require-default-props
  events: _propTypes["default"].objectOf(_propTypes["default"].any),
  // eslint-disable-line react/require-default-props
  pathFunction: _propTypes["default"].func,
  // eslint-disable-line react/require-default-props
  abs: _propTypes["default"].number,
  innerRadius: _propTypes["default"].number,
  radius: _propTypes["default"].number
};
Slice.defaultProps = {
  abs: 0,
  innerRadius: 0,
  radius: 100
};
var _default = Slice;
exports["default"] = _default;