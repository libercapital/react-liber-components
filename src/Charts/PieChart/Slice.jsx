import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { arc } from 'd3-shape';
import { Slice as VictorySlice } from './styles';

const Slice = ({ innerRadius, radius, events, abs, ...props }) => {
  const pathFunctionBg = arc()
    .outerRadius(2 * abs + radius)
    .innerRadius(innerRadius);
  const pathFunctionFg = arc()
    .outerRadius(abs / 2 + radius)
    .innerRadius(innerRadius - abs / 2);
  const defaultPathFunction = arc()
    .outerRadius(radius)
    .innerRadius(innerRadius);
  return (
    <Fragment>
      <g {...events}>
        <VictorySlice
          {...props}
          style={{ fill: 'transparent' }}
          pathFunction={pathFunctionBg}
        />

        <VictorySlice
          {...props}
          pathFunction={abs > 0 ? pathFunctionFg : defaultPathFunction}
        />
      </g>
    </Fragment>
  );
};

Slice.propTypes = {
  slice: PropTypes.objectOf(PropTypes.any), // eslint-disable-line react/require-default-props
  style: PropTypes.objectOf(PropTypes.any), // eslint-disable-line react/require-default-props
  events: PropTypes.objectOf(PropTypes.any), // eslint-disable-line react/require-default-props
  pathFunction: PropTypes.func, // eslint-disable-line react/require-default-props
  abs: PropTypes.number,
  innerRadius: PropTypes.number,
  radius: PropTypes.number,
};
Slice.defaultProps = {
  abs: 0,
  innerRadius: 0,
  radius: 100,
};

export default Slice;
