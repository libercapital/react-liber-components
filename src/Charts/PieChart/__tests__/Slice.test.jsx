import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import Slice from '../Slice';

describe('Test for Slice', () => {
  configure({ adapter: new Adapter() });
  const props = {
    abs: 0,
    innerRadius: 0,
    radius: 100,
    style: {},
    slice: {},
    events: {},
    pathFunction: sinon.spy(),
  };
  it('should render correctly', () => {
    const wrapper = shallow(<Slice {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
