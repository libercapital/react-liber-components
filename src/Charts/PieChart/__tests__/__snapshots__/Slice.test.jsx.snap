// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Slice should render correctly 1`] = `
<Fragment>
  <g>
    <styles__Slice
      pathFunction={[Function]}
      slice={Object {}}
      style={
        Object {
          "fill": "transparent",
        }
      }
    />
    <styles__Slice
      pathFunction={[Function]}
      slice={Object {}}
      style={Object {}}
    />
  </g>
</Fragment>
`;
